sdk/logical/events.go

```diff
@@ -5,10 +5,22 @@ package logical
 
 import (
 	"context"
+	"encoding/json"
+	"errors"
 
 	"github.com/hashicorp/go-uuid"
+	"google.golang.org/protobuf/types/known/structpb"
 )
 
+// common event metadata keys
+const (
+	EventMetadataSecretPath        = "secret_path"
+	EventMetadataOperation         = "operation"
```

Comment:
It would be nice to enforce some rules on this field because it should be tightly scoped to one of a few valid values. Could we define a `type EventOperation string` and enumerate all the valid options for this? And then validate them as we receive events in the core's event bus?

Reply:
I think it is better to leave metadata to be more free-form, so that plugins can define and use whatever fields they want. (Hence why we're using a generic `structpb` and not a fixed protobuf `message`.

Reply:
My motivating use-case for this comment was the RFC discussion where our consumers need to know whether the secret data has changed and they therefore need to re-fetch it. I don't think this field satisfies that requirement unless we have a schema that they can rely on for interpreting events. I'm open to other solutions too though; off the top of my head:

* Do we envisage emitting events for non-write events? If not, perhaps consumers should always assume underlying data change? Although I would worry a little bit about boxing ourselves in by committing to that implicit assumption.
* A different, more tightly scoped field name like a boolean "data_changed", or "updated", or "refetch" etc.?